packages/client-react-streaming/src/registerApolloClient.tsx

```diff
@@ -2,7 +2,21 @@ import type { ApolloClient } from "@apollo/client/index.js";
 import { cache } from "react";
 
 export function registerApolloClient(makeClient: () => ApolloClient<any>) {
-  const getClient = cache(makeClient);
+  function wrappedMakeClient() {
+    if (arguments.length) {
+      throw new Error(
+        `
+You cannot pass arguments into \`getClient\`!
+If you were to pass arguments into \`getClient\`, you would get a different copy
+of Apollo Client for every time you called it with different arguments, resulting
+in duplicate requests and a non-functional cache, essentially defeating the purpose
+of \`registerApolloClient\` and \`getClient\`.
```

Comment:
```suggestion
You cannot pass arguments into \`getClient\`.
Passing arguments to \`getClient\` returns a different copy
of Apollo Client each time its called with different arguments, potentially 
resulting in duplicate requests and a non-functional cache. This defeats 
the purpose of \`registerApolloClient\` and \`getClient\`.
```

I really like this change! I'd word this a little more using the present tense rather than future tense.

I'd also consider dropping the last part altogether since risks stated before it provide enough information on why this isn't allowed.

```suggestion
You cannot pass arguments into \`getClient\`!
If you were to pass arguments into \`getClient\`, you would get a different copy
of Apollo Client for every time you called it with different arguments, resulting
in duplicate requests and a non-functional cache.
```

Reply:
```suggestion
You cannot pass arguments into \`getClient\`.
Passing arguments to \`getClient\` returns a different copy
of Apollo Client each time its called with different arguments, potentially 
resulting in duplicate requests and a non-functional cache. 
```